In [244]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [245]:
import pandas as pd

In [246]:
df = pd.read_csv("tested.csv")


# Titanic Datasets #

ชุดข้อมูลชุดนี้เป็นที่นิยมสำหรับการวิเคราะห์ข้อมูลและเรียนรู้ของเครื่องจักร มีข้อมูลเกี่ยวกับผู้โดยสารบนเรือ Titanic ซึ่งรวมถึงลักษณะเช่น อายุ เพศ ค่าโดยสาร ห้องนอน และสถานะการรอดชีวิต เราจะทำการวิเคราะห์ข้อมูลเชิงลึก (Exploratory Data Analysis, EDA) ในชุดข้อมูล Titanic โดยใช้ Python ในโครงการนี้ 


In [247]:
df

PassengerId  Survived  Pclass  \
0            892         0       3   
1            893         1       3   
2            894         0       2   
3            895         0       3   
4            896         1       3   
..           ...       ...     ...   
413         1305         0       3   
414         1306         1       1   
415         1307         0       3   
416         1308         0       3   
417         1309         0       3   

                                             Name     Sex   Age  SibSp  Parch  \
0                                Kelly, Mr. James    male  34.5      0      0   
1                Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                       Myles, Mr. Thomas Francis    male  62.0      0      0   
3                                Wirz, Mr. Albert    male  27.0      0      0   
4    Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   
..                                            ...     ...   ...    ...    ...   
413                            Spector, Mr. Woolf    male   NaN      0      0   
414                  Oliva y Ocana, Dona. Fermina  female  39.0      0      0   
415                  Saether, Mr. Simon Sivertsen    male  38.5      0      0   
416                           Ware, Mr. Frederick    male   NaN      0      0   
417                      Peter, Master. Michael J    male   NaN      1      1   

                 Ticket      Fare Cabin Embarked  
0                330911    7.8292   NaN        Q  
1                363272    7.0000   NaN        S  
2                240276    9.6875   NaN        Q  
3                315154    8.6625   NaN        S  
4               3101298   12.2875   NaN        S  
..                  ...       ...   ...      ...  
413           A.5. 3236    8.0500   NaN        S  
414            PC 17758  108.9000  C105        C  
415  SOTON/O.Q. 3101262    7.2500   NaN        S  
416              359309    8.0500   NaN        S  
417                2668   22.3583   NaN        C  

[418 rows x 12 columns]

1. <u> จัดทำโดย </u>
</br>
</br>
6610402175 ฟาร์ฮาน่า มาเหล็ม หมู่ 1
</br>
6610405930 เบญญภา ประคุณหังสิต หมู่ 1
</br>
6610405972 พัทธนันท์ ขันตี หมู่ 1

2. แหล่งที่มา https://www.kaggle.com/datasets/brendan45774/test-file

3. แหล่งข้อมูลหลักเกี่ยวกับผู้โดยสารไททานิคมาจาก **Encyclopedia Titanica** โดยชุดข้อมูลที่ถูกใช้มาจากการรวบรวมโดยนักวิจัยหลายคน หนึ่งในแหล่งข้อมูลดั้งเดิมคือหนังสือ *"Titanic: Triumph and Tragedy" โดย Eaton & Haas (1994) Patrick Stephens Ltd และเรียบเรียงโดย Michael A. Findlay*

4. ชุดข้อมูล Titanic Datasets จัดเก็บในรูปแบบดังนี้

|Format|Binary/Text| Human-Readable|
|-----|----|----|
|CSV|Text|Yes|

5. ชุดข้อมูล Titanic Datasets โดยข้อมูลแต่ละแถว คือ ข้อมูลรายบุคคลจำนวน 418 บุคคล และข้อมุลแต่ละหลัก คือ คุณลักษณะของผู้โดยสาร 12 คุณลักษณะ

6. สิ่งที่ต้องการทำนายจากการศึกษาชุดข้อมูลนี้ คือ ศึกษาโอกาสในการรอดชีวิตของผู้ใช้งาน โดยใช้กรณีศึกษาที่ได้จากข้อมูลผู้โดยสารบนเรือไททานิก

7. สิ่งที่ใช้ในการทำนาย คือ Survived, Pclass, Sex, Age, Sibsp, Parch และ Cabin เพราะเป็นข้อมูลเชิงปริมาณ สามารถนำมาคำนวณเพื่อหาโอกาสในการรอดชีวิตของผู้ใช้งาน 

| ตัวแปร  | ความหมาย |ค่าประจำตัวแปร |
| ------------- | ------------- | ------------- |
| Survived |จำนวนผู้รอดชีวิต | 0 = เสียชีวิต, 1 = รอดชีวิต |
| Pclass | ชั้นโดยสาร 1st = ระดับสูง, 2nd = ระดับกลาง, 3rd = ระดับต่ำ |  |
| Fare | ค่าบัตรโดยสารของชั้นโดยสารแต่ละชั้น |  
| Sex | เพศสภาพ |  |
| Age | อายุ (ปี) หากน้อยกว่า 1 หากอายุถูกประเมินให้ใช้รูปแบบ xx.5 |  |
| Sibsp | จำนวนพี่น้อง / คู่สมรส, พี่น้อง = พี่น้อง, พี่เลี้ยง, น้องเลี้ยง คู่สมรส = สามี, ภรรยา (โดยไม่รวมถึงชู้สาวและคู่หมั้น) | |
| Parch | จำนวนบิดามารดา / บุตร, บิดามารดา = แม่, พ่อ บุตร = ลูกสาว, ลูกชาย, บุตรสาวเลี้ยง, บุตรชายเลี้ยง (หากระบุไว้ว่าเดินทางคนเดียวกับพี่เลี้ยง ดังนั้น ไม่มีข้อมูล parch สำหรับพวกเขา)|  |
| Cabin | หมายเลขห้องพัก |  |



8. ในการจัดเตรียมข้อมูลต้องตัดข้อมูล NaN ออก สามารถทำได้โดย
```
df = df.dropna()
df
```

9. ข้อมูลที่ไม่น่าจะได้ใช้ในระบบต้นแบบ คือ Name, Fare, Embarked และ PassengerId เพราะเป็นข้อมูลเชิงคุณภาพ ทำให้ไม่สามารถนำมาคำนวณได้

| ตัวแปร  | ความหมาย |
| ------------- | ------------- | 
| Name | ชื่อ |  
| Embarked | ท่าเรือที่ผู้โดยสารขึ้นเรือ| 
| PassengerId | หมายเลขประจำตัวผู้โดยสาร| 

10. ข้อมูลชุดนี้ไม่ควรนำไปจัดเก็บในรูปแบบอื่นเนื่องจากการจัดเก็บข้อมูลในรูปแบบตารางจะทำให้ง่ายต่อการอ่านและสรุปข้อมูล

11. ประเด็นอื่นๆ ที่เกี่ยวข้องกับข้อมูล : เนื่องจากข้อมูลชนิดนี้มีการเผยแพร่จากหลากหลายแหล่งข้อมูล

12. การมีส่วนร่วมของสมาชิกในกลุ่ม : เยี่ยมจริงๆๆ

13. การเปิดเผยการใช้เครื่องมือปัญญาประดิษฐ์ : -